java-checks/src/main/java/org/sonar/java/checks/unused/UnusedPrivateFieldCheck.java

```diff
@@ -242,15 +243,20 @@ private static List<JavaTextEdit> computeExpressionCaptures(List<AssignmentExpre
     return edits;
   }
 
-  private static String computeReplacement(ExpressionTree variable, int index) {
+  private String computeReplacement(ExpressionTree variable, int index) {
     String name = "";
+    IdentifierTree identifier;
     if (variable.is(Tree.Kind.IDENTIFIER)) {
-      name = ((IdentifierTree) variable).name() + index;
-    } else if (variable.is(Tree.Kind.MEMBER_SELECT)) {
-      name = ((MemberSelectExpressionTree) variable).identifier().name() +  index;
+      identifier = ((IdentifierTree) variable);
+      name = identifier.name() + index;
+    } else {
+      identifier = ((MemberSelectExpressionTree) variable).identifier();
+      name = identifier.name() +  index;
     }
     name = Character.toUpperCase(name.charAt(0)) + name.substring(1);
-    return String.format("var valueFormerlyAssignedTo%s = ", name);
+    TypeTree typeInDeclaration = ((VariableTree) identifier.symbol().declaration()).type();
+    String type = QuickFixHelper.contentForTree(typeInDeclaration, context);
+    return String.format("%s valueFormerlyAssignedTo%s = ", type, name);
```

Comment:
I think this approach will fail on the following example:
```
private class QuickfixAndCall {
    private int x;

    QuickfixAndCall(int y) {
      System.out.println(x = y + 1);
    }
}
```

The solution might be to declare the new variable one line before its usage?

Reply:
Actually the rule do not trigger here, so it is all good.